### DISCUSSION TITLE
bring the Slingbox to the Echo Show

### DISCUSSION THREAD

**Discussion by RocketAlf** [GerryDazoo/Slinger#505] [General]
I have an application here that may be of interest to others. 

I run several Amazon Echo devices (Echo Show 1st Gen and Echo Show 15). I have long been looking for a way to display the Slingbox output on the Echo Show. Now I have found a solution ...monoclecam.com. The service is still in beta and therefore (still) free. The service is actually used to display local IP cameras with RTSP interface on Echo Show devices.  The Slingbox is then a kind of "camera".  

On the Echo side, this requires the smart home skill "Monocle RTSP/RTP IP cameras". 
The cameras can then be set up on the monoclecam.com website. There is good documentation on how to proceed with the setup. 
Additional SW components are required for the setup. 
Via the "monocle-gatway" the local IP address of the stream source (IP camera) is passed to Echo Show, which then establishes the connection to the stream source.

Unfortunately, there seems to be so much "freedom" in the implementation of the RTSP protocol, and Echo Show only accepts a very specific protocol variant. Therefore, direct connection with the IP camera is often not possible and certainly not with the Slinger server. Additional SW components are needed. I have had success with the "rtsp-simple-server" (https://github.com/aler9/rtsp-simple-server). Echo Show, even the newer Echo Show 15, accept the interface provided by this RTSP server.

According to the configuration rules of the server, all stream sources can then be configured as input. On the output side, the IP address of the RTSP server and the respective stream name are set as source on monoclecam.com.  
But there is still a "small problem". Slinger does not deliver an RTSP stream. Therefore the HTTP stream from Slinger has to be converted into a RTSP stream. For this I use FFMPEG. In my case the corresponding entry in the configuration file of the RTSP server looks like this:

paths:
  stream6:
    source: publisher
    runOnDemand: ffmpeg -loglevel quiet -i http://192.168.2.252:9090/Slingbox/M1tv -vcodec libx264 -f rtsp -rtsp_transport tcp rtsp://localhost:8554/stream6
    runOnDemandStartTimeout: 20s
    runOnDemandCloseAfter: 10s
    #runOnDemandRestart: yes

Note: The stream names (in the example "stream6") in the should always be chosen the same. 

On monoclecam.com the following was entered:
 
Name: Slingbox TV
Stream Url: rtsp://192.168.2.140:8554/stream6
Video Codec: H264
Audio Codec: AAC
Video Resolution640*480
Tags: @tunnel

The command "Alexa, show Slingbox TV" will then display the stream of the Slingbox "M1tv" on the screen of the Echo Show.  The Slingbox "M1tv" is connected to a TV that cannot be controlled by the Slingbox remote. This is not a big deal for me, because I just want to have the same TV program available in several rooms at the same time (living room, kitchen, winter garden, ...).

This works very well when the RTSP server (along with the FFMPEG conversion) is running on my Windows PC. But when the RTSP server is running on a Raspberry Pi, there are constant interruptions for "loading" :-(. I have tried different variations of the settings, but without success. The Raspi seems to be not powerful enough for this task. I have tested Raspberry 2 and 3. 
  
Maybe one or the other already had this idea and got similar results (?). 

Because of the performance problem with the Raspi the question **@Gerry**: Could Slinger also provide an RTSP interface?
**@All** the question: Do you have experiences to bring the Slingbox to the Echo Show? Maybe also with completely different approaches (?).

I am grateful for any hint and help to make my project a success :-).  

Translated with www.DeepL.com/Translator (free version)  :-o

**Comment by robosmith** [comment]
Pi 3 is barely powerful enough to send streams from one Pro HD. Per Gerry, Pro HD requires extra processing.
Pi 4 is powerful enough to handle 2 Pro HD streams, so you might want to try that, depending on how much extra processing load your RTSP processing requires.

**Reply by GerryDazoo** [reply]
Robo, while I was working on the next release I found that there had been some debug code left in the data path. Not sure how much it would impact the real-time but if you want I can send you the latest code to see if it helps.

But buddy's got an M1 so he should be good to go. The overhead is minimal for the 500/350/M1s compared to ProHD.

**Reply by robosmith** [reply]
@GerryDazoo I would appreciate any improved code you could provide. Thanks.

**Reply by RocketAlf** [reply]
I also noticed that the ProHD runs a bit "stuttery", while the M1 is transferred without problems. I have the slinger-server running on a Raspberry Pi 2.  Can we expect this to improve in the next upgrade?

By the way, when can we expect the official version 4? I already have the pre-release version 4 running now. I like it very much, especially the SlingerPlayer. It would be great if it was also available for Android.

**Reply by RocketAlf** [reply]
**@Gerry**: Just asking again...slinger-server does not provide RTSP access (video H.264 and audio AAC) directly and will not in the future? That would be the requirement to transmit directly to the Echo Show without all the intermediary servers and transcoders.

**Reply by Toei79** [reply]
i was in that situation i end using one thinkcentre for linux server. 

i feel stream from slingbox its very picky , idk if you know gerry did a independent player instead vlc, a windows player for slingbox more friendly with the streams coming from it , so thas the situation,  

i was trying use on tivimate i never get it, only through xteve

**Reply by GerryDazoo** [reply]
RocketELF: I'm not going to do RTSP but that doesn't stop someone else from doing it.
Yes, the PI2 might not not have enough juice for the processing the stream from a ProHD, Reducing your bandwidth setting will help. The latest code does have some minor improvement in the ProHD process but I don't expect it to be significant, give it a try. I'm working on an Android version of SlingerPlayer. 
Robo: Here's the latest pre-release py.
[slingbox_server_4.0 (3).zip](https://github.com/GerryDazoo/Slinger/files/10487965/slingbox_server_4.0.3.zip)

**Reply by RocketAlf** [reply]
@Toei79: Thankyou for your tip. It seems that the Pi is not powerful enough for my task (at least not the Pi2 and the Pi3).  Using the Lenovo ThinkCenter exclusively for this purpose may work, but it seems to me like "to take a sledgehammer to crack a nut ". Furthermore, the power consumption should be quite high (?).

**Reply by Toei79** [reply]
> @Toei79: Thankyou for your tip. It seems that the Pi is not powerful enough for my task (at least not the Pi2 and the Pi3). Using the Lenovo ThinkCenter exclusively for this purpose may work, but it seems to me like "to take a sledgehammer to crack a nut ". Furthermore, the power consumption should be quite high (?).

Idk i have small one Version the one like 2.5 smartphone size, I play with other stuff on it

**Reply by RocketAlf** [reply]
@Toei79: Can you tell me what type of Lenovo ThinkCentre you are using?

**Reply by Toei79** [reply]
> @Toei79: Can you tell me what type of Lenovo ThinkCentre you are using?

Tiny

Like this one
https://www.ebay.com/itm/314348296043?mkcid=16&mkevt=1&mkrid=711-127632-2357-0&ssspo=xTCFn6ZFRgO&sssrc=2349624&ssuid=ibVD0_g0S6m&var=&widget_ver=artemis&media=COPY

How much power consumption  it can have?  

I'm running local tv channels with my files then I switched  to this one instead pi

**Reply by robosmith** [reply]
> RocketELF: I'm not going to do RTSP but that doesn't stop someone else from doing it. Yes, the PI2 might not not have enough juice for the processing the stream from a ProHD, Reducing your bandwidth setting will help. The latest code does have some minor improvement in the ProHD process but I don't expect it to be significant, give it a try. I'm working on an Android version of SlingerPlayer. Robo: Here's the latest pre-release py. [slingbox_server_4.0 (3).zip](https://github.com/GerryDazoo/Slinger/files/10487965/slingbox_server_4.0.3.zip)

@GerryDazoo have installed and used your 4.0 trial release. With limited experience, I have noticed that a power cycle reboot gives superior performance and I can watch an hour+ during prime time with NO stream drops using VLC or MxPlayer. However, after some time elapses (half day plus) streams revert to dropping every few minutes during prime time, with regular brief interruptions under more favorable Internet conditions. The other day, this phenomena seemed like a switch being flipped: solid reliable stream for 1/2 hour suddenly became routinely disrupted. Neither app restart nor Pi reboot fixed it, but a power cycle the next day did. Not sure what could account for this behavior.

**Comment by lehmbarsingh** [comment]
For those who wants the easier way .I was able do the same thing by installing mx player pro on echo show 15 and streaming from slinger(side loading the apk). I followed youtube video (How to sideload apps on echo show 15).

**Reply by RocketAlf** [reply]
Thanks for the tip. I know the procedure and have already carried it out myself. But unfortunately it does not work for the old Echo Show (1st Gen). Works only on the Echo Show 15 because that is a combination of Fire TV and Alexa.

**Reply by RocketAlf** [reply]
One more note: It is quite laborious to launch the mx player and then also to enter or select the desired stream address. Do you have an idea how to call the mx player with the desired stream using a "one click" operation or a single voice command? For example "Alexa, open my slingbox" ;-).

**Reply by tom76017** [reply]
I'm using MX Player on my Firestick.  Take a look at https://github.com/GerryDazoo/Slinger/discussions/307 to see how to save the stream location.

**Reply by sp222222** [reply]
> For those who wants the easier way .I was able do the same thing by installing mx player pro on echo show 15 and streaming from slinger(side loading the apk). I followed youtube video (How to sideload apps on echo show 15).

what apk? mxplayer or mxplayer PRO ?  and what are the steps? do you have the YT link handy?

I have no idea what you guys are doing in this thread but I might spend 250$ and get a echoShow

**Reply by RocketAlf** [reply]
@sp222222 What we do here :-o ? It's about displaying the program of a slingbox on an Echo Show. Not a trivial matter. 
I followed the way to use Alexa's interface for (surveillance) cameras "Alexa.CameraStreamController Interface 3". This is supported by MonocleCam (monoclecam.com), so there is "only minor" effort for configuration. I have already integrated several surveillance cameras this way, so adding Slinger seemed like a small step. 
The difficulty with Slinger is that slinger-server does not provide the stream in the format required by Alexa (Echo Show). Hence the trouble with transcoding. Also, Amazon changed the interface requirements for the newer Echo Show 15, so I had to find a solution that fits both the older first generation Echo Show and the new Echo Show 15 (hence the RTSP server).

In the discussion "Lehmbarsingh" introduced a suggestion, but it works exclusively for the new Echo Show 15 (and certainly for FireTV Stick). By means of "side loading" of an APP that can display the Slinger stream. The side loading procedure is described e.g. here: "https://forum.iobroker.net/topic/60891/echo-show-15-nahezu-alle-apps-installieren" (albeit in German, but why is there deepl or google translator?). This works for mx player pro, but unfortunately not for VLC.

But since I want to watch the Slinger stream on all my Echo Show devices(even the older ones), I have to stick with the camera approach. For those who have only one Echo Show to operate, the variant with the mx player pro is certainly the better and easier solution ...but, it is "cool" to start the TV program with "Alexa, show Slingbox TV" ;-).

PS: Yes, the Echo Show 15 is comparatively very expensive :-(. But I already had him ;-)  ...as an entertainment center in the kitchen ;-)

**Reply by RocketAlf** [reply]
For those who also wanted to look at the Echo Show 15th Slinger. At the beginning of February, Amazon closed the backdoor for sideloading. The instructions using "X-plore File Manager" (e.g. https://www.aftvnews.com/how-to-sideload-apk-apps-on-the-amazon-echo-show-15/) or "TV File Manager" (e.g. https://forum.iobroker.net/topic/60891/echo-show-15-nahezu-alle-apps-installieren) no longer work because the menu item "Files" or "Dateien" under "System Apps" is no longer displayed. This means that the installation of downloaded apk files is no longer possible.

If someone knows a still working solution, please report.

**Reply by GerryDazoo** [reply]
I don't have a "Show" but I would try using ADB. Most Android boxes have a "Developer Options" where you can turn this on.

**Comment by RocketAlf** [comment]
Another (almost) pleasant news to finish ...for those who might be interested. 

I have now bought a ThinClient "HP t620" (AMD G-Series GX-415GA SOC: Quad-Core APU 1.5 GHz) for 30 euros on eBay after the tip from @Toei79. It runs quietly without a fan and has a just acceptable energy consumption of about 11 watts.  Installed a Debian Linux distribution (openmediavault, because I want to use the device in parallel as a NAS).

I have taken down the RTSP server together with the FFMPEG transcoding from Pi in the architecture described at the beginning and installed it on the "HP t620". As a result, I can now stream (but only one stream) "stutter-free" via the RTSP server with transcoding. The CPU load is 50..80 %. A second stream with transcoding is not possible...for that it needs even more computing power. I will "play" a bit with the video bitrate. Maybe then two streams with transcoding are possible in parallel.

**Reply by lehmbarsingh** [reply]
I was interested in experimenting with this rtspsimpleserver for the purpose of playing source on echo show 15 ,but i am having trouble setting up server on pc ,if someone can help me and share their configuration file .I already haver my ipcamera displaying on echo show by using Monocle. I don't have deep knowledge of commands used . I have downloaded ffmpeg.exe and placed in same folder as C:\rtsp-simple-server-main .I tried to follow the steps above but not successfully. I do have slinger running on same pc and it works good . 
paths:
stream6:
source: publisher
runOnDemand: ffmpeg -loglevel quiet -i http://192.168.2.252:9090/Slingbox/M1tv -vcodec libx264 -f rtsp -rtsp_transport tcp rtsp://localhost:8554/stream6
runOnDemandStartTimeout: 20s
runOnDemandCloseAfter: 10s
#runOnDemandRestart: yes

Note: The stream names (in the example "stream6") in the should always be chosen the same.

On monoclecam.com the following was entered:

Name: Slingbox TV
Stream Url: rtsp://192.168.2.140:8554/stream6
Video Codec: H264
Audio Codec: AAC
Video Resolution640*480
Tags: @tunnel

this is my config
paths:
stream6:
source: publisher
                                                                                 this is my pc ip running slinger my main pc
runOnDemand: ffmpeg -loglevel quiet -i http://192.168.1.233:64642/mytv/slingm1med -vcodec libx264 -f rtsp -rtsp_transport tcp rtsp://localhost:8554/stream6
runOnDemandStartTimeout: 20s
runOnDemandCloseAfter: 10s
#runOnDemandRestart: yes

Note: The stream names (in the example "stream6") in the should always be chosen the same.

On monoclecam.com the following was entered:

Name: Slingbox TV
Stream Url: rtsp://192.168.1.233:8554/stream6
Video Codec: H264
Audio Codec: AAC
Video Resolution640*480
Tags: @tunnel

**Reply by GerryDazoo** [reply]
Unless you changed your URL base in your config file then
http://192.168.2.252:9090/Slingbox/M1tv  should be http://192.168.2.252:9090/slingbox/M1tv 
Lower case "s" on slingbox.

**Reply by lehmbarsingh** [reply]
My URL base in slinger is http://192.168.1.233:64642/mytv/slingm1med ,this is what use to watch using mx player

**Reply by lehmbarsingh** [reply]
Here is my config file ,I have changed the extension on it to upload. If someone can correct it. 
[rtsp-simple-server.txt](https://github.com/GerryDazoo/Slinger/files/10559683/rtsp-simple-server.txt)

**Reply by RocketAlf** [reply]
@Gerry I set my URL base to "Slinger" (upper case 'S'). Can that cause problems? I also use "Remote" (upper case 'R' for the remote und it works well. Should I change something?

**Reply by GerryDazoo** [reply]
Nope, you're good to go.

**Comment by RocketAlf** [comment]
Hello @lehmbarsingh,
try it in this way. All properties and commands concerning the "stream6" (you can also name the stream differently) must be in the section of the stream "stream6" ...so "source" and the "runOnDemand" commands and options. Please delete the "runOnDemand" commands in the "all" section. Also you should disable all other protocols if you don't use them ...set rtmpDisable, hlsDisable, webrtcDisable to "yes".  Please change the section as described:

...
paths:

  stream6:
     source: publisher
     runOnDemand: ffmpeg -loglevel quiet -i http://192.168.1.233:64642/mytv/slingm1med -vcodec libx264 -f rtsp -rtsp_transport tcp rtsp://localhost:8554/stream6
     runOnDemandStartTimeout: 20s
     runOnDemandCloseAfter: 10s
     #runOnDemandRestart: yes
    
   all:
     # Source of the stream. This can be:
     # * publisher -> the stream is published by a RTSP or RTMP client
     # * rtsp://existing-url -> the stream is pulled from another RTSP server / camera
    ...etc. etc.

A few more notes:
- If slinger is running on the same PC/board as the RTSP server, then you can replace the IP address "192.168.1.233" with "localhost" or "127.0.0.1".
- I assume that you have installed Monocle correctly. For Echo Show 15 the installation of Monocle Gateway is mandatory (for this also the tag "@tunnel"). This is all well described in the Monocle documentation. For the older generations of Echo Show, the "Proxy Server Workaround for Incompatible Cameras" (Live555 Proxy Server solution) also worked. I still use this for my old Echo Show devices and for the input side connection of other "incompatible cameras" to the RTSP server.
- I am not sure if the option "runOnDemandRestart: yes" really makes sense. I still have it commented out and everything runs just fine. If you see frequent interruptions of the stream, then it might make sense to remove the comment '#'.
- If you start the RTSP server "manually" in a dos box, then you can follow the messages of the server. And if you remove the option "-loglevel quiet" from the ffmpeg command line, you can also see the messages from ffmpeg. This can help to detect possible errors.
- You can also install ffmpeg in any other directory. However, the full path to ffmpeg must be included in the command line. But be careful, with Windows you have to set a double backslash  instead of a single for the backslash as directory separators (because in the notation YAML (YAML Ain't Markup Language) of the yml file the backslash  represents the introduction of a special character. 
Therefore, also the indentations must also be observed (similar to Python).

I hope it helped and wish you good luck. 

In the preview, the indentations disappear. Therefore I attach the code snippet here:
[snippet.txt](https://github.com/GerryDazoo/Slinger/files/10562797/snippet.txt)

**Reply by lehmbarsingh** [reply]
I will give a try tomorrow .thanks for your help

**Comment by RocketAlf** [comment]
Who can help? As I wrote before I bought a "HP t620" (AMD G-Series GX-415GA SOC: Quad-Core APU 1.5 GHz) and installed openmediavault (OMV, openmediavault.org) as NAS operating system based on Debian. I also installed slinger and the RTSP server to connect to the Echo Show 15.

Now the problem.   When I install Flask because of the remote control function, suddenly configuration changes and SW installations using apt/apt-get no longer works. It throws an error message ""ImportError: cannot import name 'Markup' from 'jinja2′". This is due to an incompatibility between Flask and Jinja2. There is an extensive discussion about this in the relevant forums on the Internet. However, I did not manage to fix the error. To make the NAS function(s) of OMV work completely, the only possibility was to uninstall Flask again.
@All: Has anyone experienced this situation and possibly been able to fix the error?
@Gerry: Flask consists of several components (packages) that can also be installed individually:
pip install --upgrade **babel**
pip install --upgrade **python-dateutil**
pip install --upgrade **flask-moment**
pip install --upgrade **flask-wtf**
pip install --upgrade **flask_sqlalchemy**   
Are all these packages required for the remote control to work? 
Or, could another WEB server be used (e.g. NGINX)?

**Reply by GerryDazoo** [reply]
Don't modify the version of python the system is using. Use the Python environment capabilities for your own private version of python to use. 
https://docs.python.org/3/library/venv.html
This is exactly what I've done on my NAS.

**Reply by RocketAlf** [reply]
Many, many thanks. 
I did not know this option yet. Have followed the "instruction manual" ...and it works wonderfully now :-)))

**Reply by Toei79** [reply]
Wondering if this can be used to play in other devices?

**Comment by lehmbarsingh** [comment]
Hello@RocketAlf , Seems like I am missing on something because I haven't been able to get it working .I followed all your steps or directions .I have disabled all protocols you mentioned not being used ( rtmpDisable, hlsDisable, webrtcDisable to "yes" ) .I have installed ffmpeg properly . I have created bat files to run the server automatically .so 1st I run the server then I open ffmpeg . now if I try opening the stream6 using vlc (rtsp://127.0.0.1:8554/stream6) it should start streaming  but it never happens. Can you share your config file at lehmbar@yahoo.ca .Also if you can give me step by step how to run this properly would be appreciated .when I give the command in ffmpeg (ffmpeg -loglevel quiet -i http://192.168.1.233:64642/mytv/slingm1med -vcodec libx264 -f rtsp -rtsp_transport tcp rtsp://localhost:8554/stream6) manually it connects to my slinger but after few seconds it cuts off. If I understand correctly when anyone whether echo show 15 or vlc player try to connect to stream6 it suppose to start basically  start converting hls stream to rtsp . Thanks

**Reply by RocketAlf** [reply]
Now it becomes a bit difficult ;-). I attach below my full yml configuration file of the rtsp-simple-server.

When you have everything set up:
1. activate MonocleCam skill for your Amazon account.
2. on monoclecam.org configure rtsp-simple-server (stream6) as camera:
Name: Slingbox TV
Stream URL: rtsp://192.168.1.233:8554/stream6
Video codec: H264
Audio codec: AAC
Video resolution 640*480
Tags:@tunnel
("192.168.1.233" is the address of the device running the RTSP server)
3. add the camera "Slingbox TV" to the Alexa app on your smart phone.
Create the correct yml configuration file for the RTSP server. In the section "stream6:" must be at least:
source: publisher
runOnDemand: ffmpeg -loglevel quiet -i http://127.0.0.1:64642/mytv/slingm1med -vcodec libx264 -f rtsp -rtsp_transport tcp rtsp://localhost:8554/stream6
("127.0.0.1 "assuming Slinger is running on the same device as the RTSP server,  otherwise the full IP address of Slinger).
5. open DOS-box. Change to the directory where the RTSP server is installed. Start the RTSP server. Then the DOS-box will show some messages about the startup process.
6. Note: Do NOT start ffmpeg manually!!! ffmpeg is started by the RTSP server according to the runOnDemand command when the stream is retrieved from the RTSP server. ffmpeg has no server function. Therefore, VLC cannot retrieve the stream from ffmpeg directly, but must use the "detour" via the RTSP server.
7. Test: start VLC and under network open stream execute the following command: rtsp://localhost:8554/stream6 (assuming VLC is running on the same device as RTSP server, otherwise rtsp://192.168.1.233:8554/stream6.
The DOS box now shows that the RTSP server has received the request and the stream is being transmitted. The RTSP server now starts ffmpeg according to the runOnDemand option and connects as "receiver" for the stream with ffmpeg and passes it on as "stream6". If additionally "-loglevel quiet" was removed from the ffmpeg call, the messages of ffmpeg will appear here as well (quite a mess ;-).
If everything has worked so far, then go to the Echo Show.
10. and say the command "Alexa, show Slingbox TV".

Now after some time the stream should appear on the screen. It may not work the first time. Then repeat the command. 

Does it work ?

**Reply by RocketAlf** [reply]
[rtsp-simple-server.txt](https://github.com/GerryDazoo/Slinger/files/10610066/rtsp-simple-server.txt)

**Reply by RocketAlf** [reply]
One more addition. I assume that the Monocle Gateway has been correctly installed on your PC according to the instructions on monoclecam.com.

**Reply by RocketAlf** [reply]
see also for Monocle setup:
https://www.hackster.io/monocle/monocle-view-control-ip-cameras-with-alexa-arduino-32bdd3

**Reply by lehmbarsingh** [reply]
Yes , Finally it worked. I would like to thank everyone for help .I do have multiple echo show device in the house including 8 inch and 10 and the 15. It's working on all of these devices thru voice command  by calling show my slingbox tv . I couldn't figure why the rtsp server was not getting my command from vlc or smplayer .I rebooted my pc and made some changes to my router and then it suddenly started working. I did notice that my dell optiplex 9020 with i7 and nividia gpu is seems to consume around 20% of cpu power . Is there any way to get ffmpeg to use nividia gpu for encoding and decoding ? I am also interested if possible to stream to web page so i access at work thru internet . Thanks again for your help.

**Reply by lehmbarsingh** [reply]
Now that I have one of the slingbox running with one source ,I would like have multiple slingboxes(3) running thru this rtsp server with different sources and connecting to multiple clients .what is proper way to config in yaml file.Do I need to setup multiple servers .

**Reply by RocketAlf** [reply]
This is not a problem. For each slingbox (for each source) a stream must be configured. So the first slingbox is "stream6". The second one is for example "stream7" and the third one "stream8". You can of course define the stream names as you like).

In Monocle (monoclecam.com) you have to define two additional "cameras":

Name: Slingbox Basement
Stream Url: rtsp://192.168.1.233:8554/stream7
Video Codec: H264
Audio Codec: AAC
Video Resolution: 640*480
Tags: @tunnel

Name: Slingbox TV Roof
Stream Url: rtsp://192.168.1.233:8554/stream8
Video Codec: H264
Audio Codec: AAC
Video Resolution: 640*480
Tags: @tunnel

("192.168.1.233:8554": IP and Port of the RTSP server. 
The value "Video Resolution: 640*480" depends on the setting for the Slingbox. I have set the value "Resolution=5" in Slingers config.ini. If you set the value differently you have to change "Video Resolution" accordingly).  

Register the two new "cameras" in the Alexa app.

In the yml configuration file of the RTSP server now add the sections for the two new slingboxes, for example:

stream7:

> source: publisher

> runOnDemand: ffmpeg -loglevel quiet -i http://127.0.0.1:64642/mytv/sb2name -vcodec libx264 -f rtsp -rtsp_transport tcp rtsp://localhost:8554/stream7

stream8:

> source: publisher

> runOnDemand: ffmpeg -loglevel quiet -i http://127.0.0.1:64642/mytv/sb3name -vcodec libx264 -f rtsp -rtsp_transport tcp rtsp://localhost:8554/stream8

("sb2name, sb3name the names of your two new slingboxes, defined in Slinger's config.ini, 
"127.0.0.1 "assuming the one Slinger server is running on the same device as the RTSP server, otherwise the full IP address of Slinger. 
But you can also set up a second or even third Slinger server. Then you have to adjust IP addresses and/or port numbers and/or URL bases accordingly.)

Restart RTSP server ...and enjoy ;-)

Note: But there is the following to consider. The transcoding process of ffmpeg is very power consuming ...and also streaming from the ProHD). Depending on the computing power of the device, this may limit the number of streams that can be played simultaneously.

**Reply by lehmbarsingh** [reply]
Thank you Again for your help. Everything is working great. I would like to thank everyone who contributed to these projects . I really enjoy these projects. Love learning new things everyday. Thanks.

**Reply by lehmbarsingh** [reply]
I have another question concerning slinger server , is there any way we can watch slingbox stream directly using chrome like we used to back in days. Now the only option  I have is to use rtsp simple server and use ffmpeg and convert the stream to rtsp and watch using chrome browser.

**Reply by GerryDazoo** [reply]
No you can't. Why do you feel compelled to view using Chrome?
I don't think you could every watch a Slingbox using Chrome. Or do you mean using a browser to launch the proprietary SlingPlayer for Web plugin?

**Reply by lehmbarsingh** [reply]
Hi GerryDazoo , When I try your full version of slingerplayer at my work pc apparently it doesn't work it gives me error of "is not a valid integer value" but at home pc I don't get that error and it work perfectly. Is it possible that work pc are protected by security. presently I am able to watch using chrome browser on work pc using rtsp simple server running at home . but because the ffmpeg is very power consuming .  I do have multiple Slingboxes running at home . Yes I guess it was SlingPlayer for Web Plugin. My understanding is that do you absolutely need to convert http stream to rtsp in order to view by web browser is there way get rtsp directly from slingbox. BTW thanks to you for us to reuse our slingboxes. 
  stream1:
source: publisher
runOnDemand: ffmpeg -i http://192.168.1.233:64642/mytv/m1v1sd -vcodec libx264 -f rtsp -rtsp_transport tcp rtsp://192.168.1.233:8554/stream1
runOnDemandStartTimeout: 20s
runOnDemandCloseAfter: 10s

**Reply by RocketAlf** [reply]
Hello Lehmbarsingh, 
honestly, I don't quite understand the question. you write: "presently I am able to watch using chrome browser on work pc using rtsp simple server running at home .". Why so cumbersome? Using "rtsp simple server" is only required to create an rtsp stream for watching on an Echo Show. If you watch on a pc, then you can directly use the http stream of the SlingerServer. (Almost) every video player will play the http stream (e.g. also VLC). So why do you want to use the rtsp stream for watching on a PC?
Gerry wrote earlier in response to a question of mine that the SlingerServer will not provide an rtsp stream. I can understand that too.

**Reply by lehmbarsingh** [reply]
The problem I have is at my work we have access to internet , but none of programs to stream slingboxes work at work pc .(vlc portable or slingerplayer or SMplayer portable. The only option I have is go thru rtsp server. I am able to watch on my cell phone with all of above but due to limited data I have .

**Reply by RocketAlf** [reply]
So ...if none of the programs mentioned work, then I rather suspect that there is something wrong with the configuration of the firewall of your home network. Your SlingerServer is located in your home network on "http://192.168.1.233:64642/mytv/m1v1sd". Did you set the port forwarding on the router for port 64642 correctly? Only then the SlingerServer can be reached from outside. I assume that you have a global IP URL for your network (e.g. using Dyn-DNS service). But if the rtsp stream works, then the port forwarding for port 8554 must be set correctly (?).  Hmmm ...strange !?!? 

There you still have something to tinker ...good luck :-)

**Reply by lehmbarsingh** [reply]
My port forwarding must be  done right because i can watch it from my phone outside of home network. I am thinking it's most likely problem on work pc ,we are not allowed install any programs. work pc are locked to admin. But I was trying with portable vlc and your slingerplayer.

**Reply by GerryDazoo** [reply]
It's quite possible that the IT people at your work have blocked outgoing access to web-sites not on their approved list.

**Reply by lehmbarsingh** [reply]
That's what I am thinking of but at least I can still watch thru http://myip:8888/stream2 .

**Reply by RocketAlf** [reply]
@lehmbarsingh: 
1. you have enabled HLS protocol and use http on port 8888 (?)
2. transcoding to H.264 and rtsp protocol is then actually superfluous.  I know it works, but there might be a better setting.
3. what is your setting for stream2 ?
4. if the port forwarding of port 8888 works, then the port forwarding of port 64642 must work, too, if you didn't make a (typo) mistake in the configuration (!). 

Maybe try another port number for the SlingerServer ( less than 49152 ). The port 64642 belongs to a range which is used by Apple for the "Xsan filesystem access". This may cause problems within the company network.

This is really puzzling :-O !?!?!

**Reply by lehmbarsingh** [reply]
1. yes I have enable HLS and I do use port 8888 . I have configured the ports properly in my Asus router.
2. transcoding to H.264 and rtsp protocol is then actually superfluous. I know it works, but there might be a better setting.
  what is best setting you recommend
3. stream2:
     source: publisher
     runOnDemand: ffmpeg -i http://192.168.1.233:64642/mytv/m1v1hd -vcodec libx264 -f rtsp -rtsp_transport tcp rtsp://192.168.1.233:8554/stream2
     runOnDemandStartTimeout: 20s
     runOnDemandCloseAfter: 10s
4. I will give it try with lower port as you mentioned. 
  Well Known Ports: 0 through 1023.
  Registered Ports: 1024 through 49151.
  Dynamic/Private : 49152 through 65535.

**Reply by lehmbarsingh** [reply]
Here is setting for slingbox M1 
[m1v1hd]
sbtype="350/500/M1"
password=E1:DB1CAB5D2003A43B37A1426A0F817DE75E16F2532F26F03142A9E7745C578E12
;finderID 1E850B582B0CD3B61F3309474D489CA1
ipaddress=192.168.1.143
port=5304
Resolution=16
FrameRate=30
VideoBandwidth=3000
VideoSmoothness=63
VideoSource=1
maxstreams=6
Remote=bell3.txt

**Reply by lehmbarsingh** [reply]
[SERVER]
; local port number for the server to listen on for connections
; Send all streaming and Remote request to this port number.
port=45452
maxremotestreams=6
URLbase=mytv
enableremote=yes
